tasks/environments-pipeline.yaml

```diff
@@ -0,0 +1,142 @@
+                - name: deploy_image_name
+                  value: $(tasks.pr-configuration.results.deploy-image-name)
+                - name: init_overlay_contextpath
+                  value: $(tasks.init-task-run.results.overlay_contextpath)
+                - name: overlay_name
+                  value: $(tasks.init-task-run.results.overlay_name)
+                - name: overlay_contextpath
+                  value: $(tasks.pr-configuration.results.overlay-contextpath)
+                - name: default-action
+                  value: {1}
+                - name: default_branch
+                  value: $(params.default_branch)
+                - name: registry
+                  value: $(params.registry)
+                - name: registry_org
+                  value: $(params.registry)
+                - name: registry_proejct
+                  value: $(params.registry)
+              resources:
+                - name: s2i-thoth
+                  resourceSpec:
+                    type: image
+                    params:
+                      - name: url
+                        value: $(resources.inputs.s2i-thoth.url)
+            """
+            taskrun.format(env, env.default-action)
+            output = open(f"taskruns/{env}-taskrun.yaml")
+            output.write(taskrun)
+            output.close()
+
+    - name: overlays-pipeline-setup
+      image: $(resources.inputs.s2i-thoth.url)
+      workingDir: /workspace/repo
+      securityContext:
+        privileged: true
+      script: |
+        #!/usr/bin/env python3
+        import os
+        import json
+        import yaml
+        import uuid
+
+        ci_conf, thoth_conf = {}, {}
+        with open(".aicoe-ci.yaml", "r") as ci_stream:
+            ci_conf = yaml.safe_load(ci_stream)
+        uid = str(uuid.uuid4())[0:6]
+        for env in ci_conf:
+            with open(f"/workspace/taskruns/{env}-taskrun.yaml", "r") as taskrun_yaml:
+                taskrun = yaml.safe_load(taskrun_yaml)
+                spec_params = []
+                for param in taskrun["spec"]["params"]:
+                    for ele in ci_conf.get("deploy", []):
+                        if param["name"] == "overlay_contextpath":
+                            param["value"] = overlay_env["deploy"].get(
+                                "overlay-contextpath", ""
+                            )
+                        elif param["name"] == "deploy_" + "_".join(ele.split("-")):
+                            param["value"] = overlay_env["deploy"].get(ele, "")
+                    if param["name"] == "deploy":
+                        param["value"] = "deploy" if overlay_env.get("deploy") else ""
+                    if param["name"] == "overlays_name":
+                        param["value"] = overlay_name
+                    if param["name"] == "overlays_dir":
+                        param["value"] = overlays_dir
+                    spec_params.append(param)
```

Comment:
This is only looping through deploy, we want to loop from the elements in the deploy, need extra loop